target_chains/sui/contracts/sources/pyth_accumulator.move

```diff
@@ -0,0 +1,157 @@
+        // and passed to this function as input.
+        let vaa_size = deserialize::deserialize_u16(cursor);
+        let _ = deserialize::deserialize_vector(cursor, (vaa_size as u64));
+
+        let merkle_root_hash = parse_accumulator_merkle_root_from_vaa_payload(vaa_payload);
+        parse_and_verify_accumulator_updates(cursor, merkle_root_hash, clock)
+    }
+
+    // parse_accumulator_merkle_root_from_vaa_payload takes in some VAA bytes, verifies that the vaa
+    // corresponds to a merkle update, and finally returns the keccak hash of the merkle root.
+    // Note: this function is adapted from the Aptos Pyth accumulator
+    fun parse_accumulator_merkle_root_from_vaa_payload(message: vector<u8>): Bytes20 {
+        let msg_payload_cursor = cursor::new(message);
+        let payload_type = deserialize::deserialize_u32(&mut msg_payload_cursor);
+        assert!(payload_type == ACCUMULATOR_UPDATE_WORMHOLE_VERIFICATION_MAGIC, E_INVALID_WORMHOLE_MESSAGE);
+        let wh_message_payload_type = deserialize::deserialize_u8(&mut msg_payload_cursor);
+        assert!(wh_message_payload_type == 0, E_INVALID_WORMHOLE_MESSAGE); // Merkle variant
+        let _merkle_root_slot = deserialize::deserialize_u64(&mut msg_payload_cursor);
+        let _merkle_root_ring_size = deserialize::deserialize_u32(&mut msg_payload_cursor);
+        let merkle_root_hash = deserialize::deserialize_vector(&mut msg_payload_cursor, 20);
+        cursor::take_rest<u8>(msg_payload_cursor);
+        bytes20::new(merkle_root_hash)
+    }
+
+    // Note: this parsing function is adapted from the Aptos Pyth parse_price_feed_message function
+    fun parse_price_feed_message(message_cur: &mut Cursor<u8>, clock: &Clock): PriceInfo {
+        //let message_cur = cursor::new(message);
+        let message_type = deserialize::deserialize_u8(message_cur);
+
+        assert!(message_type == 0, 0); // PriceFeedMessage variant
+        let price_identifier = price_identifier::from_byte_vec(deserialize::deserialize_vector(message_cur, 32));
+        let price = deserialize::deserialize_i64(message_cur);
+        let conf = deserialize::deserialize_u64(message_cur);
+        let expo = deserialize::deserialize_i32(message_cur);
+        let publish_time = deserialize::deserialize_u64(message_cur);
+        let _prev_publish_time = deserialize::deserialize_i64(message_cur);
+        let ema_price = deserialize::deserialize_i64(message_cur);
+        let ema_conf = deserialize::deserialize_u64(message_cur);
+        let price_info = price_info::new_price_info(
+            clock::timestamp_ms(clock) / 1000, // not used anywhere kept for backward compatibility
+            clock::timestamp_ms(clock) / 1000,
+            price_feed::new(
+                price_identifier,
+                pyth::price::new(price, conf, expo, publish_time),
+                pyth::price::new(ema_price, ema_conf, expo, publish_time),
+            )
+        );
+        //cursor::rest(message_cur);
```

Comment:
Why have you commented this out? We want to consume the rest for future compatibility.

Reply:
It's because `&mut message_cur` is passed into this function, but I can also pass in the actual message_cur, instead of destroying it on the outside.
```
            let message_cur = cursor::new(message);
            let price_info = parse_price_feed_message(&mut message_cur, clock);
            cursor::destroy_empty(message_cur);
```